tooling/profiler/src/cli/opcodes_flamegraph_cmd.rs

```diff
@@ -31,59 +33,47 @@ pub(crate) struct OpcodesFlamegraphCommand {
 pub(crate) fn run(args: OpcodesFlamegraphCommand) -> eyre::Result<()> {
     run_with_generator(
         &args.artifact_path,
-        &InfernoFlamegraphGenerator { count_name: "opcodes".to_string() },
         &args.output,
+        &InfernoFlamegraphGenerator { count_name: "opcodes".to_string() },
         args.skip_brillig,
     )
 }
 
 fn run_with_generator<Generator: FlamegraphGenerator>(
     artifact_path: &Path,
-    flamegraph_generator: &Generator,
     output_path: &Path,
+    flamegraph_generator: &Generator,
     skip_brillig: bool,
 ) -> eyre::Result<()> {
+    info!("Reading program from {:?}", artifact_path);
+
     let mut program =
-        read_program_from_file(artifact_path).context("Error reading program from file")?;
+        read_program_from_file(artifact_path).with_context(|| "Could not read program from file")?;
 
     let acir_names = std::mem::take(&mut program.names);
     let brillig_names = std::mem::take(&mut program.brillig_names);
 
     let bytecode = std::mem::take(&mut program.bytecode);
+    let debug_artifact = DebugArtifact::from(program);
 
-    let debug_artifact: DebugArtifact = program.into();
+    // There's one debug information per source file
+    assert_eq!(debug_artifact.debug_symbols.len(), 1);
```

Comment:
This is not correct and will break on program which use the `#[fold]` attribute.